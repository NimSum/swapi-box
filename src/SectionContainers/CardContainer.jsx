import React from 'react';
import PropTypes from 'prop-types';
import Crawl from 'react-star-wars-crawl';
import { CharacterCard } from '../components/Cards/CharacterCard';
import { PlanetCard } from '../components/Cards/PlanetCard';
import { VehicleCard } from '../components/Cards/VehicleCard';

export const CardContainer = ({ opening_crawl, title, release_date, episode_id, cards, category, updateFavoriteCount }) => {
  const romanNumeral = ['I', 'II', 'III', 'IV', 'V', 'VI', 'VII', 'VII']
  const crawl = (
    < Crawl
      title={ `Episode ${romanNumeral[episode_id - 1]}` }
      subTitle={ title }
      text={ opening_crawl }
      releaseYear={ release_date }
    />)
  
  return !cards.length
    ? (<div className="crawl-container">{crawl}</div>)
    : (<section className="card-container">
        { category === 'people' && 
          < CharacterCard cards={ cards } 
            updateFavoriteCount={updateFavoriteCount} /> }
        { category === 'planets' && 
          < PlanetCard cards={ cards } 
            updateFavoriteCount={updateFavoriteCount} /> } 
        { category === 'vehicles' && 
          < VehicleCard cards={ cards } 
            updateFavoriteCount={updateFavoriteCount} /> }
      </section>)
}

CardContainer.propTypes = {
  opening_crawl: PropTypes.string,
  title: PropTypes.string,
  release_date: PropTypes.string,
  episode_id: PropTypes.number,
  cards: PropTypes.array,
  category: PropTypes.string
}