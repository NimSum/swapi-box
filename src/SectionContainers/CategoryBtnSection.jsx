import React, { Component } from 'react'
import { PlanetsBtn, PeopleBtn, VehiclesBtn, ViewFavoritesBtn } from '../components/Buttons/Buttons';

export class CategoryBtnSection extends Component {
  constructor() {
    super();
    this.state = {
      activeBtn: '',
    }
  }

  setActiveBtn = e => {
    this.setState({ activeBtn: e.target.name });
    this.props.changeCategory(e.target.name);
  }

  render() {
    const { activeBtn } = this.state;

    return (
      <section className="categories-btn-container">
        <div className="btn-container">
          < PlanetsBtn 
            setActive={ this.setActiveBtn }
            activeBtn={ activeBtn } />
          < PeopleBtn 
            setActive={ this.setActiveBtn }
            activeBtn={ activeBtn } />
          < VehiclesBtn 
            setActive={ this.setActiveBtn }
            activeBtn={ activeBtn } />
          < ViewFavoritesBtn 
            setActive={ this.setActiveBtn }
            activeBtn={ activeBtn } 
            favoriteCount={ this.props.favoriteCount } />
        </div>
      </section>
    )
  }
}

export default CategoryBtnSection
