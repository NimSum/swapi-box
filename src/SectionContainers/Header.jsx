import React from 'react';

export const Header = ({ category }) => {
  return (
    <header>
      <img src={require('../images/star-wars-title.png')} alt="Star Wars Title"/>
      { category && <h2>{ category.toUpperCase() }</h2>}
    </header>
  )
}
