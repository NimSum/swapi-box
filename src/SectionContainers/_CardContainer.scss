section.card-container {
  border-radius: 20px;
  width: 70%;
  display: grid;
  margin: auto;
  justify-content: center;
  grid-gap: 6% 4%;
  grid-template-columns: repeat(auto-fill, minmax(300px, 350px));
  height: 70vh;
  overflow: scroll;
}

div.crawl-container {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 70%;
}
