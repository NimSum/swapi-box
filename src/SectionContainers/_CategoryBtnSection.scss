section.categories-btn-container {
  position: absolute;
  bottom: 0;
  width: 100%;
  padding: 20px; 
  height: 17vh;
  border: 1px dashed white;

  div.btn-container {
    width: 50%;
    margin: auto;
    display: flex;
    justify-content: center;

  }
}