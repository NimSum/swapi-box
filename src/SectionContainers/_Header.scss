header {
  margin: 0;
  padding: 20px 0;
  color: $star-wars-font-color;
  font-family: $content-font;


  h2 {
    margin-top: 20px;
  }
  img {
    width: 300px;
    height: 100px;
  }
}