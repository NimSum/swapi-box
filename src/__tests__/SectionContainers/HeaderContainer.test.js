import { shallow } from 'enzyme';
import React from 'react';
import { Header } from '../../SectionContainers/Header';

describe('Header', () => {
  
  it('Should match snapshot', () => {
    let wrapper = shallow(< Header />);

    expect(wrapper).toMatchSnapshot();
  })

})