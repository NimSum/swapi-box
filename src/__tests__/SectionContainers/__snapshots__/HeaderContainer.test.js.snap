// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Should match snapshot 1`] = `
<header>
  <img
    alt="Star Wars Title"
    src="star-wars-title.png"
  />
</header>
`;
